### PR TITLE
Friendships v1

### PR DESCRIPTION
**Pull request by atoledo-ec**
## Friendships v1

In this branch we added the model and controller for friendship between users.
Also, we create and modify the views to display the friendships and the pending friendships between users.

### CODE REVIEW

**Review by andresporras3423 — CHANGES_REQUESTED** [review]
### Status: Changes Required :recycle:
**Hello Alejandro Toledo Freire** 
Good job with your first submission for this project :muscle:  :muscle:
It meets most of the requirements :clap: :clap: :clap:  
There are just a few details to improve before you can go ahead:

- [x] Add a screenshot in your README file
- [x] Add spec tests to your project
- [x] In Friendship class, both sender and receiver should be associated with User class
 
Please request another review once you are ready! :+1: 
 
Happy coding :relaxed:
 
Oscar Andrés Russi Porras

File: app/models/user.rb [hunk 1]
```diff
@@ -9,4 +9,35 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+  has_many :friendships
+  has_many :inverse_friendships, class_name: :Friendship, foreign_key: :friend_id
```
Comment (ghost) [inline app/models/user.rb:13]: Write spec tests for the above associations. Hint you can use [shoulda matchers](https://matchers.shoulda.io/) to write single line tests.

File: app/models/user.rb [hunk 2]
```diff
@@ -9,4 +9,35 @@ class User < ApplicationRecord
...
+  def friend?(user)
+    friends.include?(user)
+  end
+
+  def friends
+    friends_array = friendships.map { |friendship| friendship.friend if friendship.confirmed }
+    friends_array += inverse_friendships.map { |friendship| friendship.user if friendship.confirmed }
+    friends_array.compact
+  end
 end
```
Comment (ghost) [inline app/models/user.rb]: Write spec tests for the above methods. If you have any methods that you are not using outside the User class, you can declare them private and avoid writing tests for them.

**Review by ghost — CHANGES_REQUESTED** [review]
# Status: Changes Required 🤓

### Hello team 😄,

**Great job implementing the requested changes 💪. However, there are still a few fixes needed.**

1). **You only wrote spec tests for two files in `app/models` directory. Add tests for all the other files as well.**

2). **Add tests for your `helpers` and ` views`.**

3). **You have `rubocop` error.**
rubocop

4). **Check more in the comments below.**

5).  **The screenshot images in your readme file are broken**.

### Happy coding 🥂 

**PS: You can reach me on slack (Rose Wanjohi).**

**Comment by atoledo-ec** [comment]
I checked for rubocop issues in Actions section but I got nothing

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
## Status: Changes Required! :recycle:
### Hi @alejandrotoledoweb and @Blaise-Shyaka :clap: :clap:
#### You have done an incredible job writing such a comprehensive integration test. As much as I'm eager to approve this project some of the required changes from the last review were not implemented. Check my comments below :point_down:

#### Tests:
- **_Write tests for public methods in the User model as mentioned in the last review_**

- **_Write association tests for the User model(similar to the association test you wrote in the friendship spec) as mentioned in the last review_**

#### _Reviewed by Temitayo Adeleke_

**Review by andresporras3423 — CHANGES_REQUESTED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:
**Hello Alejandro Toledo Freire** 
Amazing job :tada: :tada: :tada:  you have met all the requirements for this project
Now you are ready to continue with your next challenge :muscle: 
Don't forget to merge with the development branch
 
Keep going like this  :clap: :clap:
 
Oscar Andrés Russi Porras

**Pull request merged**